File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n=len(nums)
        if n==0:
            return 0
        nums_set=set(nums)
        longest=0
        for item in nums_set:
            if item-1 not in nums_set:
                lenght=1
                while item+lenght in nums_set:
                    lenght+=1
                longest=max(longest,lenght)
        return longest

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int longest=0;
        for(int num:nums){
            set.add(num);
        }
        for(int num:set){
            if(!set.contains(num-1)){
                int length=1;
                int next_num=num+1;
                while(set.contains(next_num)){
                    length+=1;
                    next_num+=1;
                }
                longest=Math.max(longest,length);
                System.out.println(num);
            }
        }
        System.out.println(set.toString());
        return longest;
        
    }
}